src/Language/Futhark/Parser/Lexer.x

```diff
@@ -51,12 +53,12 @@ import Futhark.Util.Loc hiding (L)
 @unop = "!"
 @qualunop = (@identifier ".")+ @unop
 
-@opchar = ("+"|"-"|"*"|"/"|"%"|"="|"!"|">"|"<"|"|"|"&"|"^"|".")
-@binop = ("+"|"-"|"*"|"/"|"%"|"="|"!"|">"|"<"|"|"|"&"|"^") @opchar*
+$opchar = [\+\-\*\/\%\=\!\>\<\|\&\^\.]
+@binop = ($opchar # \.) $opchar*
 @qualbinop = (@identifier ".")+ @binop
 
 @space = [\ \t\f\v]
-@doc = "-- |"[^\n]*(\n@space*"--"[^\n]*)*
```

Comment:
Can you explain why this change is right?  Won't this match the following?

```
-- | foo bar
baz
```

Or does `.` not match newlines in Alex?

Reply:
Yes – in Alex (like many regexes) `.` is everything but newlines.

[From the docs:](https://www.haskell.org/alex/doc/html/charsets.html)
"The built-in set ‘.’ matches all characters except newline (\n)."